### PR TITLE
WT-8879 Set the OOO flag when the selected tombstone is globally visible (5.0 backport) (#7632)

### PR DESCRIPTION
**Pull request by etienneptl**
(cherry picked from commit 23e5746ea948e3b1ac903f2cce3ea209e19329a1)

### CODE REVIEW

**Review by lukech — APPROVED** [review]
LGTM

File: src/reconcile/rec_visibility.c [hunk 1]
```diff
@@ -717,16 +717,27 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, W
      * Set the flag if the selected tombstone is an out-of-order or mixed mode to an update. Based
      * on this flag, the caller functions perform the history store truncation for this key.
      */
-    if (tombstone != NULL &&
+    if (!is_hs_page && tombstone != NULL &&
       !F_ISSET(tombstone, WT_UPDATE_RESTORED_FROM_DS | WT_UPDATE_RESTORED_FROM_HS)) {
         upd = upd_select->upd;
         while (upd != NULL && upd->txnid == WT_TXN_ABORTED)
```
Comment (kommiharibabu) [inline src/reconcile/rec_visibility.c:723]: This while loop is not required anymore.

**Review by kommiharibabu — CHANGES_REQUESTED** [review]
Unnecessary extra while loop which doesn't require.

File: src/reconcile/rec_visibility.c (hunk 1 above)
Comment (etienneptl) [reply src/reconcile/rec_visibility.c]: Good catch, fixed in 7807909b18e53c22a6dd9a9b2da4c8d6fa54687b

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Pull request merged**